webapp/src/pages/applicationForm/components/ApplicationForm.js

```diff
@@ -15,7 +15,9 @@ import FormReferenceRequest from "./ReferenceRequest";
 import Loading from "../../../components/Loading";
 import _ from "lodash";
 
+
 const baseUrl = process.env.REACT_APP_API_URL;
+let errorMessage = false;
```

Comment:
The introduction of global state makes me uncomfortable, can you explain the logic behind this and is there an alternative that avoids global state?

Reply:
Ive made changes here, I studied the flow of the entire js File more in depth and I found a more conventional approach, please review. Still getting familiar with the codebase but its making more sense everyday